src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,387 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Linq.Expressions;
+
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Formatter;
+using Microsoft.AspNet.OData.Formatter.Serialization;
+using Microsoft.AspNet.OData.Query.Expressions;
+using Microsoft.OData;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData.Query
+{
+    /// <summary>
+    /// Default implementation of SkipTokenHandler for the service. 
+    /// </summary>
+    public class DefaultSkipTokenHandler : SkipTokenHandler
+    {
+        private const char CommaDelimiter = ',';
+        private static char propertyDelimiter = ':';
+        internal static  DefaultSkipTokenHandler Instance = new DefaultSkipTokenHandler();
+
+        /// <summary>
+        /// Constructor for DefaultSkipTokenHandler - Sets the Property Delimiter
+        /// </summary>
+        public DefaultSkipTokenHandler()
+        {
+            IsDeltaFeedSupported = false;
+        }
+
+        /// <summary>
+        /// Constructor for Unit testing purposes - Sets the Property Delimiter
+        /// </summary>
+        public DefaultSkipTokenHandler(char delimiter)
```

Comment:
>public DefaultSkipTokenHandler(char delimiter) [](start = 8, length = 46)

If that's for unit testing, shall we make it internal?

Reply:
E2E test cases are not in the same project as this. It will become inaccessible. 

---
In reply to: [282612090](https://github.com/OData/WebApi/pull/1724#discussion_r282612090) [](ancestors = 282612090)